### PR TITLE
First working version

### PR DESCRIPTION
**Pull request by dwantula**

### CODE REVIEW

File: src/App.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+body {
+background-color: gainsboro;
```
Comment (mi-chal) [inline src/App.scss]: Wrong indentation

File: src/screens/ToDoList/ToDoList.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import { saveItemInLocalStorage, getItemFromLocalStorage } from '../../shared/utils/LocalStorage';
+import './styles.scss';
+
+class ToDoListComponent extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: [],
+      newItem: '',
+
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Unnecessary space

File: src/screens/ToDoList/ToDoList.jsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    }
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Missing semicolon

File: src/screens/ToDoList/ToDoList.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    this.onInputChange = this.onInputChange.bind(this);
+    this.onAddNewItemClick = this.onAddNewItemClick.bind(this);
+    this.markCompleted = this.markCompleted.bind(this);
+    this.deleteItem = this.deleteItem.bind(this);
+    this.deleteAllItems = this.deleteAllItems.bind(this);
+  }
+
+  componentDidMount() {
+    const tasks = getItemFromLocalStorage('items') || [];
+    this.setState({items: tasks})
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Missing semicolon. Object should have additional spaces - `{ items: tasks }`

File: src/screens/ToDoList/ToDoList.jsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  onInputChange(inputValue) {
+    this.setState({ newItem: inputValue })
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Missing semicolon

File: src/screens/ToDoList/ToDoList.jsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+      items: [],
+      newItem: '',
+
+    }
+
+    this.onInputChange = this.onInputChange.bind(this);
+    this.onAddNewItemClick = this.onAddNewItemClick.bind(this);
+    this.markCompleted = this.markCompleted.bind(this);
+    this.deleteItem = this.deleteItem.bind(this);
+    this.deleteAllItems = this.deleteAllItems.bind(this);
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx:21]: Why do you bind all these methods?

File: src/screens/ToDoList/ToDoList.jsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+  onAddNewItemClick() {
+    const { items, newItem } = this.state;
+    const itemToAdd = { id: generateId('item'), isCompleted: false, text: newItem};
+    const newItems = [...items, itemToAdd];
+    this.setState({
+      items: newItems ,
+      newItem: ''
+    });
+    saveItemInLocalStorage('items', newItems)
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Missing semicolon

File: src/screens/ToDoList/ToDoList.jsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+   }
+
+  markCompleted(id) {
+    const index = this.state.items.findIndex(item => item.id === id)
+    const newItems = [...this.state.items]
+    const itemStatus = newItems[index].isCompleted;
+    console.log(itemStatus)
+    newItems[index].isCompleted = !itemStatus
+    this.setState({items: newItems})
+    saveItemInLocalStorage('items', newItems)
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Missing semicolons

File: src/screens/ToDoList/ToDoList.jsx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  deleteItem(id) {
+    const newItems = this.state.items.filter(element => element.id !== id )
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Unnecessary space at the end. Missing semicolon.

File: src/screens/ToDoList/ToDoList.jsx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+    this.setState({items: newItems})
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Missing semicolon. Object should have additional spaces.

File: src/screens/ToDoList/ToDoList.jsx [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+    saveItemInLocalStorage('items', newItems)
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Missing semicolon

File: src/screens/ToDoList/ToDoList.jsx [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  deleteAllItems() {
+    this.setState({items: []})
+    localStorage.clear('items')
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: From now if I write:
@1 it means "missing semicolon", 
@2 it means object should have additional spaces

File: src/screens/ToDoList/ToDoList.jsx [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  render() {
+    return (
+      <div className="container">
+        <h2 className="title">TO DO LIST</h2>
+        <div className="row">
+          <InputComponent   type="text" value={this.state.newItem} onChange={this.onInputChange}/>
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Unnecessary space. Move props to separate lines.

File: src/screens/ToDoList/ToDoList.jsx [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+            className="button-add-thing"
+            type="button"
+            text="Add new item"
+            onClick={this.onAddNewItemClick.bind(this)}
+          />
+        </div>
+        <div>
+          {this.state.items.map(item => (
+            <div key= {item.id} >
+              <ToDoListItemComponent markClicked={this.markCompleted} deleteClicked={this.deleteItem} item= {item} />
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Move props to separate lines

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+import TextComponent from '../../../shared/components/Input/Text/Text';
+import ButtonComponent from '../../../shared/components/Input/Button/Button';
+import './styl.scss';
+
+
+
+function ToDoListItemComponent(props) {
+  return (
+    <div className="row">
+    <div className= {` ${props.item.isCompleted ? 'row-item-completed' : 'row-item'}`} >
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Unnecessary space

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+      <ButtonComponent 
+        type= "button" 
+        className= "check-button"
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Unnecessary space

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+        onClick={() => props.markClicked(props.item.id)} 
+    />
+      <TextComponent className="row-item-completed"text= {props.item.text} />
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Unnecessary space

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+    </div>   
+      <ButtonComponent 
+        type= "button" 
+        className= "delete-button"
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Unnecessary spaces

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+        text="X"
+        onClick={() => props.deleteClicked(props.item.id)} 
+    />
+
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Unnecessary line

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+    </div>   
+      <ButtonComponent 
+        type= "button" 
+        className= "delete-button"
+        text="X"
+        onClick={() => props.deleteClicked(props.item.id)} 
+    />
+    
+    </div>
+
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Unnecessary line

File: src/screens/ToDoList/ToDoListItem/styl.scss [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+@mixin button-check-delete {
+  border-radius: 50%;
+  outline: 0px;
+  cursor: pointer;
+  margin-left: 5px;
+  height: 17px;
+  width: 17px;
+}
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/styl.scss]: The naming of this mixin is not unfortunate - it points to very specific class when mixins should be generic.

File: src/screens/ToDoList/ToDoListItem/styl.scss [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+  position: relative;
+  outline: 0;
+  cursor: pointer;
+  opacity: .5;
+  text-decoration: line-through; 
+  margin-left: 50px; 
+  align-items: center;
+}
+
+.row{
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/styl.scss]: Missing space

File: src/screens/ToDoList/ToDoListItem/styl.scss [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+
+.delete-button:hover {
+  background-color: red;
+}
+
+.check-button {
+  @include button-check-delete;
+  border: 1px solid green;
+  margin-bottom: 15px;
+  background-color: green;
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/styl.scss]: You should create separate file where you can keep all colors and later just import them anywhere.

File: src/screens/ToDoList/styles.scss [hunk 24]
```diff
@@ -0,0 +1,78 @@
+  $main-color : #40e0d0;
```
Comment (mi-chal) [inline src/screens/ToDoList/styles.scss]: Wrong indentation

File: src/screens/ToDoList/styles.scss [hunk 25]
```diff
@@ -0,0 +1,78 @@
...
+.title {
+  align-items: center;
+  text-align: center;
+  color: lightslategray;
+  font-stretch: expanded;
+  background-color: $main-color;
+  letter-spacing: 2px;
+}
+
+.clear-button {
```
Comment (mi-chal) [inline src/screens/ToDoList/styles.scss:37]: You should create `button` mixin and put there all basic styles for button like `cursor`, `outline` etc.

File: src/shared/components/Input/Input/Input.jsx [hunk 26]
```diff
@@ -0,0 +1,21 @@
...
+import './styles.scss'
+
+class InputComponent extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    this.props.onChange(event.target.value)
```
Comment (mi-chal) [inline src/shared/components/Input/Input/Input.jsx]: Missing semicolon

File: src/shared/components/Input/Input/Input.jsx [hunk 27]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  render() {
+    return (
+      <input className="input-add-item" placeholder="Write down what you want to do..." type={this.props.type} value={this.props.value} onChange={this.handleChange}></input>
```
Comment (mi-chal) [inline src/shared/components/Input/Input/Input.jsx]: Move attributes to separate lines

File: src/shared/components/Input/Input/styles.scss [hunk 28]
```diff
@@ -0,0 +1,18 @@
...
+  color: darkgray;
+  width: 290px;
+  outline: 0;
+  margin-left: 10px;
+  letter-spacing: 2px;
+  transition: transform 0.5s ease-in-out;
+}
+
+input:focus
+{
```
Comment (mi-chal) [inline src/shared/components/Input/Input/styles.scss]: Wrong {} notation

**Review by mi-chal — CHANGES_REQUESTED** [review]

File: package.json [hunk 29]
```diff
@@ -3,9 +3,15 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^1.2.32",
+    "@fortawesome/free-brands-svg-icons": "^5.15.1",
+    "@fortawesome/free-regular-svg-icons": "^5.15.1",
+    "@fortawesome/free-solid-svg-icons": "^5.15.1",
+    "@fortawesome/react-fontawesome": "^0.1.11",
```
Comment (mi-chal) [inline package.json:10]: Documentation suggests to install only `fontawesome-svg-core`, `free-solid-svg-icons`, `react-fontawesome`. Wouldn't that be enough?

File: src/App.scss [hunk 30]
```diff
@@ -0,0 +1,3 @@
+body {
+  background-color: gainsboro;
```
Comment (mi-chal) [inline src/App.scss]: Colour values should be kept in `colors.scss`

File: src/screens/ToDoList/ToDoList.jsx [hunk 31]
```diff
@@ -0,0 +1,100 @@
...
+      items: newItems ,
+      newItem: ''
+    });
+    saveItemInLocalStorage('items', newItems);
+   }
+
+  markCompleted(id) {
+    const index= this.state.items.findIndex(item=> item.id === id);
+    const newItems= [...this.state.items];
+    const itemStatus= newItems[index].isCompleted;
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Missing spaces before `=`

File: src/screens/ToDoList/ToDoList.jsx [hunk 32]
```diff
@@ -0,0 +1,100 @@
...
+    newItems[index].isCompleted = !itemStatus;
+    this.setState({ items : newItems });
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Unnecessary space before `:`

File: src/screens/ToDoList/ToDoList.jsx [hunk 33]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  deleteItem(id) {
+    const newItems = this.state.items.filter(element=> element.id !== id);
+    this.setState({items: newItems});
+    saveItemInLocalStorage('items', newItems);
+  }
+
+  deleteAllItems() {
+    this.setState({ items : [] });
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Unnecessary space before `:`

File: src/screens/ToDoList/ToDoList.jsx [hunk 34]
```diff
@@ -0,0 +1,103 @@
...
+  deleteAllItems() {
+    this.setState({ items: [] });
+    localStorage.clear('items');
+  }
+
+  render() {
+    return (
+      <div className= "container">
+        <h2 className= "title">TO DO LIST</h2>
+        <div className= "row">
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Unnecessary space before `=`

File: src/screens/ToDoList/ToDoList.jsx [hunk 35]
```diff
@@ -0,0 +1,103 @@
...
+          <ButtonComponent
+            className = "button-add-thing"
+            type = "button"
+            text = "+"
+            onClick = {this.onAddNewItemClick.bind(this)}
+          />
+        </div>
+        <div>
+          {this.state.items.map(item => (
+            <div key= {item.id}>
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Unnecessary space

File: src/App.scss (hunk 1 above)
Comment (dwantula) [reply src/App.scss]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 2 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 3 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 4 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 5 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Changed!

File: src/screens/ToDoList/styles.scss (hunk 24 above)
Comment (dwantula) [reply src/screens/ToDoList/styles.scss]: Done!

File: src/screens/ToDoList/styles.scss (hunk 25 above)
Comment (dwantula) [reply src/screens/ToDoList/styles.scss:37]: I will do this when code will be correct

File: src/shared/components/Input/Input/Input.jsx (hunk 26 above)
Comment (dwantula) [reply src/shared/components/Input/Input/Input.jsx]: Done!

File: src/shared/components/Input/Input/Input.jsx (hunk 27 above)
Comment (dwantula) [reply src/shared/components/Input/Input/Input.jsx]: Done

File: src/shared/components/Input/Input/styles.scss (hunk 28 above)
Comment (dwantula) [reply src/shared/components/Input/Input/styles.scss]: Done!

File: src/App.scss (hunk 30 above)
Comment (dwantula) [reply src/App.scss]: I will do this when code will be correct

File: src/screens/ToDoList/ToDoList.jsx (hunk 31 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 32 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 33 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 34 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Chenged!

File: src/screens/ToDoList/ToDoList.jsx (hunk 35 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx [hunk 36]
```diff
@@ -0,0 +1,104 @@
...
+    this.setState({ items: tasks });
+  }
+
+  onInputChange(inputValue) {
+    this.setState({ newItem: inputValue });
+  }
+
+  onAddNewItemClick() {
+    const { items, newItem } = this.state;
+    const itemToAdd = { id: generateId('item'), isCompleted: false, text: newItem};
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Missing space after `newItem`

File: src/screens/ToDoList/ToDoList.jsx [hunk 37]
```diff
@@ -0,0 +1,104 @@
...
+          <InputComponent   
+            type="text" 
+            value={this.state.newItem} 
+            onChange={this.onInputChange}
+          />
+          <ButtonComponent
+            className="button-add-thing"
+            type="button"
+            text="+"
+            onClick={this.onAddNewItemClick.bind(this)}
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: You've already bind this method in constructor. No need to do it again.

File: src/screens/ToDoList/ToDoList.jsx [hunk 38]
```diff
@@ -0,0 +1,104 @@
...
+          />
+        </div>
+        <div>
+          {this.state.items.map(item=> (
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Missing space before `=>`

File: src/screens/ToDoList/ToDoList.jsx [hunk 39]
```diff
@@ -0,0 +1,104 @@
...
+            <div key={item.id}>
+            <ToDoListItemComponent 
+              markClicked={this.markCompleted} 
+              deleteClicked={this.deleteItem} 
+              item={item} 
+            />
+            <p className="line"></p>
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx]: Wrong indentation. There's no need to add `key` to div element. It should ba added to `ToDoListItemComponent`

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx [hunk 40]
```diff
@@ -0,0 +1,33 @@
...
+import TextComponent from '../../../shared/components/Input/Text/Text';
+import ButtonComponent from '../../../shared/components/Input/Button/Button';
+import './styl.scss';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCheck} from '@fortawesome/free-solid-svg-icons';
+
+function ToDoListItemComponent(props) {
+  return (
+    <div className="row">
+      <div className={` ${props.item.isCompleted ? 'row-item-completed' : 'row-item'}`}>
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Unnecessary space before `$`

File: src/screens/ToDoList/ToDoListItem/styl.scss [hunk 41]
```diff
@@ -0,0 +1,50 @@
...
+  margin-top: 5px;
+  padding: 0px;
+  margin-left: 14px;
+  @include button-styles;
+}
+
+.delete-button:hover {
+  background-color: white;
+  transform: scale(1.5);
+}
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/styl.scss]: You can use `scss` syntax:
```
.delete-button {
  background-color: white;
  font-size: 15px;
  border: none;
  color: red;
  margin-top: 5px;
  padding: 0px;
  margin-left: 14px;
  @include button-styles;
  &:hover {
    background-color: white;
    transform: scale(1.5);
  }
}
```

File: src/screens/ToDoList/styles.scss [hunk 42]
```diff
@@ -0,0 +1,75 @@
...
+  border-radius: 18px;
+  width: 400px;
+  min-height: 400px;
+  margin-left: auto;
+  margin-right: auto;
+  margin-top: 100px;
+  background-color:white;
+  display: flex;
+  flex-direction: column;
+  box-shadow: 5px 8px 5px 8px gray, 30px 10px 5px -90px black;
```
Comment (mi-chal) [inline src/screens/ToDoList/styles.scss:13]: This shadow looks really bad. It should be more subtle.

File: src/screens/ToDoList/styles.scss [hunk 43]
```diff
@@ -0,0 +1,75 @@
+@import './../../shared/components/main-styles.scss';
+
+.container {
+  border: 3px solid $main-color;
+  border-radius: 18px;
+  width: 400px;
```
Comment (mi-chal) [inline src/screens/ToDoList/styles.scss]: Should be wider.

File: src/screens/ToDoList/ToDoListItem/styl.scss [hunk 44]
```diff
@@ -0,0 +1,50 @@
...
+
+.checkbox :hover {
+  transform: scale(1.5);
+}
+
+.checkbox {
+  color: green;
+  margin-right: 14px;
+  margin-top: 5px;
+  transition: transform 0.5s ease-in-out;
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/styl.scss]: This transition looks really bad.

File: src/screens/ToDoList/ToDoListItem/styl.scss [hunk 45]
```diff
@@ -0,0 +1,50 @@
...
+  color: red;
+  margin-top: 5px;
+  padding: 0px;
+  margin-left: 14px;
+  @include button-styles;
+}
+
+.delete-button:hover {
+  background-color: white;
+  transform: scale(1.5);
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoListItem/styl.scss]: This transform looks really bad.

File: src/screens/ToDoList/styles.scss [hunk 46]
```diff
@@ -0,0 +1,75 @@
...
+.text-decoration {
+  color:black;
+  display: flex;
+  letter-spacing: 2px;
+}
+
+.line {
+  border-bottom: 2px solid $main-color;
+  margin: 0 15px;
+  box-shadow: 1px 1px 1px 1px lightgray, 0px 0px 0px 0px lightgray;
```
Comment (mi-chal) [inline src/screens/ToDoList/styles.scss]: Please read about using box-shadow

File: src/screens/ToDoList/styles.scss [hunk 47]
```diff
@@ -0,0 +1,75 @@
...
+  color: $title-color;
+  font-stretch: expanded;
+  background-color: $main-color;
+  letter-spacing: 2px;
+}
+
+.clear-button {
+  width: 100px;
+  height: 30px;
+  margin: 13px 0px 10px 250px;
```
Comment (mi-chal) [inline src/screens/ToDoList/styles.scss]: You should use different approach to position button to the right. Setting margin to 250px is a bad idea.

File: src/screens/ToDoList/styles.scss [hunk 48]
```diff
@@ -0,0 +1,75 @@
...
+  border-radius: 18px;
+  border: 2px solid red;
+  @include button-styles;
+  background-color: white;
+  transition: all 0.5s ease-in-out;
+}
+
+.clear-button:hover {
```
Comment (mi-chal) [inline src/screens/ToDoList/styles.scss:51]: Use scss syntax with parent selector

File: src/screens/ToDoList/styles.scss [hunk 49]
```diff
@@ -0,0 +1,75 @@
...
+  background-color:  red;
+  box-shadow: 0px 0px 0px 0px black, 0px 0px 10px 2px black;
+  transform: scale(1.1) rotate(5deg);
```
Comment (mi-chal) [inline src/screens/ToDoList/styles.scss]: This looks VERY bad

File: src/screens/ToDoList/styles.scss [hunk 50]
```diff
@@ -0,0 +1,75 @@
...
+  height: 35px;
+  font-size: 25px;
+  margin-left: 30px;
+  @include button-styles;
+  background-color: white;
+  margin-right: 10px;
+  transition: transform 0.5s ease-in-out;
+}
+
+.button-add-thing:hover {
```
Comment (mi-chal) [inline src/screens/ToDoList/styles.scss:72]: Use scss parent selector syntax

File: src/screens/ToDoList/styles.scss [hunk 51]
```diff
@@ -0,0 +1,75 @@
...
+.clear-button:hover {
+  background-color:  red;
+  box-shadow: 0px 0px 0px 0px black, 0px 0px 10px 2px black;
+  transform: scale(1.1) rotate(5deg);
+}
+
+.button-add-thing {
+  border: 2px solid $main-color;
+  border-radius: 50%;
+  color: darkslategray;
```
Comment (mi-chal) [inline src/screens/ToDoList/styles.scss]: This color should be stored with all other colors

File: src/shared/components/Input/Input/styles.scss [hunk 52]
```diff
@@ -0,0 +1,19 @@
...
+.input-add-item {
+  border: 2px solid lightgray;
+  border-radius: 8px;
+  height: 30px;
+  color: $input-text-color;
+  width: 290px;
+  outline: 0;
+  margin-left: 10px;
+  letter-spacing: 2px;
+  transition: transform 0.5s ease-in-out;
```
Comment (mi-chal) [inline src/shared/components/Input/Input/styles.scss]: This looks terrible.

File: src/screens/ToDoList/ToDoList.jsx (hunk 37 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 38 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx (hunk 40 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Fixed!

File: src/screens/ToDoList/ToDoListItem/styl.scss (hunk 44 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/styl.scss]: done!

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx (hunk 18 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Done!

File: src/screens/ToDoList/ToDoListItem/styl.scss (hunk 45 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/styl.scss]: Cheked!

File: src/screens/ToDoList/styles.scss (hunk 47 above)
Comment (dwantula) [reply src/screens/ToDoList/styles.scss]: Checked!

File: src/screens/ToDoList/ToDoList.jsx (hunk 6 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx:21]: Because I understood it this way.

File: src/screens/ToDoList/ToDoList.jsx (hunk 7 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 8 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 9 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 10 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 11 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 12 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Ok

File: src/screens/ToDoList/ToDoList.jsx (hunk 13 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx (hunk 15 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Done!

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx (hunk 16 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Done!

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx (hunk 17 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Done!

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx (hunk 19 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Fixed

File: src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx (hunk 20 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/ToDoListItem.jsx]: Fixed!

File: src/screens/ToDoList/ToDoList.jsx (hunk 36 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Done!

File: src/screens/ToDoList/ToDoList.jsx (hunk 14 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx]: Fixed!

File: src/screens/ToDoList/ToDoListItem/styl.scss (hunk 21 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/styl.scss]: Done!

File: src/screens/ToDoList/ToDoListItem/styl.scss (hunk 22 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/styl.scss]: Done!

File: src/screens/ToDoList/ToDoListItem/styl.scss (hunk 23 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoListItem/styl.scss]: Done!

File: src/screens/ToDoList/styles.scss (hunk 25 above)
Comment (dwantula) [reply src/screens/ToDoList/styles.scss:37]: Done!

File: src/screens/ToDoList/styles.scss (hunk 48 above)
Comment (dwantula) [reply src/screens/ToDoList/styles.scss:51]: Done!

File: src/screens/ToDoList/styles.scss (hunk 43 above)
Comment (dwantula) [reply src/screens/ToDoList/styles.scss]: Done!

File: src/screens/ToDoList/styles.scss (hunk 51 above)
Comment (dwantula) [reply src/screens/ToDoList/styles.scss]: Done!

File: src/screens/ToDoList/styles.scss (hunk 50 above)
Comment (dwantula) [reply src/screens/ToDoList/styles.scss:72]: Done!

File: src/shared/components/Input/Input/styles.scss (hunk 52 above)
Comment (dwantula) [reply src/shared/components/Input/Input/styles.scss]: Fixed!

File: package.json (hunk 29 above)
Comment (dwantula) [reply package.json:10]: I don't know

File: src/screens/ToDoList/styles.scss (hunk 46 above)
Comment (dwantula) [reply src/screens/ToDoList/styles.scss]: Done!

File: src/screens/ToDoList/styles.scss (hunk 42 above)
Comment (dwantula) [reply src/screens/ToDoList/styles.scss:13]: Done!

File: src/screens/ToDoList/ToDoList.jsx [hunk 53]
```diff
@@ -0,0 +1,105 @@
...
+    this.setState({ items: tasks });
+  }
+
+  onInputChange(inputValue) {
+    this.setState({ newItem: inputValue });
+  }
+
+  onAddNewItemClick() {
+    const { items, newItem } = this.state;
+    const itemToAdd = {id: generateId('item'), isCompleted: false, text: newItem};
```
Comment (mi-chal) [inline src/screens/ToDoList/ToDoList.jsx:35]: Missing spaces in object notation

File: src/screens/ToDoList/styles.scss [hunk 54]
```diff
@@ -0,0 +1,78 @@
...
+  border-radius: 50%;
+  color: $white;
+  width: 40px;
+  height: 40px;
+  font-size: 25px;
+  margin-left: 30px;
+  @include button-styles;
+  background-color: $positive-color;
+  margin-right: 10px;
+  // transition: all 0.3s;
```
Comment (mi-chal) [inline src/screens/ToDoList/styles.scss:68]: Remove comment

**Review by mi-chal — APPROVED** [review]

File: src/screens/ToDoList/ToDoList.jsx (hunk 53 above)
Comment (dwantula) [reply src/screens/ToDoList/ToDoList.jsx:35]: Done!

File: src/screens/ToDoList/styles.scss (hunk 54 above)
Comment (dwantula) [reply src/screens/ToDoList/styles.scss:68]: Done!

**Pull request merged**